(billing) Add overview and introduce Clerk Billing 🎉 (#2200)
clerk/clerk-docs: docs/billing/overview.mdx

## Enable Billing

To enable Billing for your application, navigate to the [**Billing Settings**](https://dashboard.clerk.com/last-active?path=billing/settings) page in the Clerk Dashboard. This page will guide you through enabling Billing for your application. If your application was created after the Billing launch, you will only need to create a plan. If your application predates the Billing launch, you will need to complete the steps outlined.

## Create a plan

Plans are what your users subscribe to. There is no limit to the number of Plans you can create.

There are two types of Plans:

- Organization Plans: plans that only [organizations](/docs/organizations/overview) can subscribe to.
- User Plans: plans that only users can subscribe to.

To create a Plan, navigate to the [**Plans**](https://dashboard.clerk.com/last-active?path=billing/plans) page in the Clerk Dashboard. Here, you can create, edit, and delete Plans. When creating a Plan, you can also create Features for the Plan; see the next section for more information.

> [!TIP]
> What is the **Visible on Components** option?
>
> ---
>

> > Plans appear in some on Clerk components depending on what kind of plan it is. All Plans can appear in the `<PricingTable />` component. If it's a User Plan, it can appear in the `<UserProfile />` component. If it's an Organization Plan, it can appear in the `<OrganizationProfile />` component. When creating or editing a Plan, if you'd like to hide it from appearing in Clerk components, you can toggle the **Visible on Components** option off.

Reviewer: ```suggestion
> Plans appear in some Clerk components depending on what kind of plan it is. All Plans can appear in the `<PricingTable />` component. If it's a User Plan, it can appear in the `<UserProfile />` component. If it's an Organization Plan, it can appear in the `<OrganizationProfile />` component. When creating or editing a Plan, if you'd like to hide it from appearing in Clerk components, you can toggle the **Visible on Components** option off.
```

## Add features to a plan

Features make it easy to give entitlements to your Plans. You can add any number of Features to a Plan.

You can add a Feature to a Plan when you are creating a Plan. To add it after a Plan is created:

1. Navigate to the [**Plans**](https://dashboard.clerk.com/last-active?path=billing/plans) page in the Clerk Dashboard.
1. Select the Plan you'd like to add a Feature to.
1. In the **Features** section, select **Add Feature**.

> [!TIP]
> What is the **Visible on Components** option?
>
> ---
>

> > Plans appear in some on Clerk components depending on what kind of plan it is. All Plans can appear in the `<PricingTable />` component. If it's a User Plan, it can appear in the `<UserProfile />` component. If it's an Organization Plan, it can appear in the `<OrganizationProfile />` component. When adding a Feature to a Plan, it will also automatically appear in the corresponding Plan. When creating or editing a Feature, if you'd like to hide it from appearing in Clerk components, you can toggle the **Visible on Components** option off.

Reviewer: ```suggestion
> Plans appear in some Clerk components depending on what kind of plan it is. All Plans can appear in the `<PricingTable />` component. If it's a User Plan, it can appear in the `<UserProfile />` component. If it's an Organization Plan, it can appear in the `<OrganizationProfile />` component. When adding a Feature to a Plan, it will also automatically appear in the corresponding Plan. When creating or editing a Feature, if you'd like to hide it from appearing in Clerk components, you can toggle the **Visible on Components** option off.
```

    import { auth } from '@clerk/nextjs/server'

    export default async function Page() {
      // Use `auth()` helper to access the `has()` method
      const { has } = await auth()

      // Use `has()` method to check if user has a Feature
      const hasPremiumAccess = has({ feature: 'premium_access' })

      if (!hasPremiumAccess)
        return <h1>Only subscribers with the Premium Access feature can access this content.</h1>

      return <h1>Our Exclusive Content</h1>
    }
    ```
  </Tab>

  <Tab>
    The following example demonstrates how to use `has()` to check if a user has a Plan.

    ```tsx {{ filename: 'app/page.tsx' }}
    import { auth } from '@clerk/nextjs/server'

    export default async function Page() {
      // Use `auth()` helper to access the `has()` method
      const { has } = await auth()

      // Use `has()` method to check if user has a Plan
      const hasBronzePlan = has({ plan: 'bronze' })

      if (!hasBronzePlan) return <h1>Only subscribers to the Bronze plan can access this content.</h1>

      return <h1>For Bronze subscribers only</h1>
    }
    ```
  </Tab>
</Tabs>

### Example: Using `<Protect />`

> The [`<Protect />`](/docs/components/protect) component protects a page or part of a page by checking if the user has been granted a specific type of access control (Role, Permission, Feature, or Plan). You can pass it a `fallback` prop that will be rendered if the user does not have the access control.

Reviewer 2: ```suggestion
The [`<Protect />`](/docs/components/protect) component protects content or even entire routes by checking if the user has been granted a specific type of access control (Role, Permission, Feature, or Plan). You can pass it a `fallback` prop that will be rendered if the user does not have the access control.
```
